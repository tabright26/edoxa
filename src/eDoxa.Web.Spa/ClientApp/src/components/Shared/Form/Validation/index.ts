import FormValidation from "./Validation";

export default FormValidation;
