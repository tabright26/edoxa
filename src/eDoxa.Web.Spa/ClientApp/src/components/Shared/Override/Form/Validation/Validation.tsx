import React from "react";
import { Alert } from "reactstrap";

const ValidationSummary = ({ error }) => (
  <Alert color="primary">
    <strong>Error:</strong> {error}
  </Alert>
);

export default ValidationSummary;
