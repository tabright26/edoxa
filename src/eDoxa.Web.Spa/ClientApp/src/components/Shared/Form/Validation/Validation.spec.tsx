import React from "react";
import renderer from "react-test-renderer";
import { Form, reduxForm } from "redux-form";
import { Provider } from "react-redux";
import Validation from "./Validation";

it("renders without crashing", () => {
  //Arrange
  const store: any = {
    getState: () => {},
    dispatch: action => {},
    subscribe: () => {}
  };

  const FormWrapper = () => (
    <Form onSubmit={() => {}}>
      <Validation error={null} />
    </Form>
  );

  const CustomForm = reduxForm({ form: "TEST_FORM" })(FormWrapper);

  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <CustomForm />
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
