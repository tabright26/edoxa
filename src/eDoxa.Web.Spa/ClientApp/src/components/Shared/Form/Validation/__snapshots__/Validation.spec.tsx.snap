// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form>
  <div
    className="alert alert-primary fade"
    role="alert"
  >
    <strong>
      Error:
    </strong>
     
  </div>
</form>
`;
